### PR TITLE
Laboratory work #7, Buntova Anastasia - 20FPL1

### PR DESCRIPTION
**Pull request by DangerousCrocodile**

### CODE REVIEW

File: lab_7_llm/main.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from pathlib import Path
 from typing import Iterable, Iterator, Sequence
 
+
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:5]: Delete

**Comment by SalnikovIgor** [comment]
Necessary implementation of loading your dataset.

**Comment by SalnikovIgor** [comment]
It's time to write the loading of your dataset.

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/assets/danetqa_example.csv:1]: Delete from PR

File: seminars/seminar_01_16_2024/try_datasets.py
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/try_datasets.py:1]: Return to the main branch

File: lab_7_llm/main.py [hunk 2]
```diff
@@ -42,6 +45,13 @@ def obtain(self) -> None:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
 
+        dataset = load_dataset(
+            'ag_news',
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Get these parameters from `AbstractRawDataImporter`

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: seminars/seminar_01_16_2024/try_datasets.py [hunk 3]
```diff
@@ -63,4 +63,4 @@ def main() -> None:
 
 
 if __name__ == '__main__':
-    main()
+    main()
```
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/try_datasets.py]: New line

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/assets/danetqa_example.csv:1]: Delete from PR

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix past comments

File: seminars/seminar_01_16_2024/assets/danetqa_example.csv [hunk 4]
```diff
@@ -1,13 +1,101 @@
-question,passage,idx,label
```
Comment (SalnikovIgor) [inline seminars/seminar_01_16_2024/assets/danetqa_example.csv:1]: Delete

File: lab_7_llm/start.py [hunk 5]
```diff
@@ -2,16 +2,26 @@
...
 
 
 @report_time
-def main() -> None:
+def main():
     """
     Run the translation pipeline.
     """
-    result = None
-    assert result is not None, "Demo does not work correctly"
```
Comment (SalnikovIgor) [inline lab_7_llm/start.py:14]: Leave these line at the end of the function

File: lab_7_llm/start.py [hunk 6]
```diff
@@ -2,16 +2,26 @@
...
+    with open(PROJECT_ROOT / 'lab_7_llm' / 'settings.json', 'r', encoding='utf-8') as settings:
+        settings = json.load(settings)
+    importer = RawDataImporter(settings['parameters']['dataset'])
+    importer.obtain()
+    processor = RawDataPreprocessor(importer.raw_data)
+    return processor.analyze()
```
Comment (SalnikovIgor) [inline lab_7_llm/start.py]: This function should not return anything

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/main.py [hunk 7]
```diff
@@ -32,6 +27,7 @@ class RawDataImporter(AbstractRawDataImporter):
     """
     A class that imports the HuggingFace dataset.
     """
+    _raw_data: DataFrame
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:30]: This parameter is already created in the class `AbstractRawDataImporter`

File: lab_7_llm/main.py [hunk 8]
```diff
@@ -42,6 +38,12 @@ def obtain(self) -> None:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
 
+        self._raw_data = load_dataset(self._hf_name, split='test').to_pandas()
+
+    @property
+    def raw_data(self) -> DataFrame:
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:44]: And this

File: lab_7_llm/main.py [hunk 9]
```diff
@@ -169,6 +233,26 @@ def _infer_batch(self, sample_batch: Sequence[tuple[str, ...]]) -> list[str]:
         Returns:
             list[str]: Model predictions as strings
         """
+        predictions = []
+
+        if isinstance(sample_batch, tuple):
+            tokens = self._tokenizer(
+                str(sample_batch[0][0]),
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py:240]: Tokenizer should work fine with any batch size

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

File: lab_7_llm/main.py (hunk 7 above)
Comment (DangerousCrocodile) [reply lab_7_llm/main.py:30]: Without this line the code cannot pass mypy checks

File: lab_7_llm/main.py (hunk 8 above)
Comment (DangerousCrocodile) [reply lab_7_llm/main.py:44]: And this

**Pull request closed without merging**